doc/packages.schema.json

```diff
@@ -0,0 +1,102 @@
+{
+  "$id": "https://aka.ms/winget-packages.schema.json",
+  "$schema": "https://json-schema.org/draft/2019-09/schema#",
+
+  "title": "winget Packages List Schema",
+  "description": "Describes a list of packages for batch installs",
+
+  "additionalProperties": true,
+
+  "properties": {
+    "WinGetVersion": {
+      "description": "Version of winget that generated this file",
+      "type": "string",
+      "pattern": "^[0-9]+\\.[0-9]\\.[0-9]$"
+    },
+
+    "CreationDate": {
+      "description": "Date when this list was generated",
+      "type": "string",
+      "format": "date-time"
+    },
+
+    "Sources": {
+      "description": "Sources from which each package comes from",
+      "type": "array",
+
+      "items": {
+        "description": "A source and the list of packages to install from it",
+        "type": "object",
+        "required": true,
+        "additionalProperties": true,
+
+        "properties": {
+          "SourceDetails": {
+            "description": "Details about this source",
+            "type": "object",
+            "required": false,
+            "additionalProperties": true,
+
+            "properties": {
+              "Name": {
+                "description": "Name of the source",
+                "type": "string",
+                "required": true
+              },
+
+              "Identifier": {
+                "description": "Identifier for the source",
+                "type": "string",
+                "required": true
+              },
+
+              "Argument": {
+                "description": "Argument used to install the source",
+                "type": "string",
+                "required": true
+              },
+
+              "Type": {
+                "description": "Type of the source",
+                "type": "string",
+                "required": true
+              }
+            }
+          },
+
+          "Packages": {
```

Comment:
I did this based on a sample @denelon gave me of what he thought a file could look like. He may have a better reasoning for doing it this way (or maybe not, not sure).

My personal take on it is that it's likely that many people may want to view, edit or even write these files by hand, and I'd imagine most of the time they would be installing many files from a few sources or even just one. In that case, maybe it would be a slightly experience to not have dozens of lines just saying `source: "winget"`. Of course, I may be completely wrong here in how this is going to be used and what the expierence would be like.